### PR TITLE
Publish sbt plugin to bintray

### PR DESCRIPTION
**Pull request by jroper**
* sbt plugin releases get published to bintray
* sbt plugin snapshots get published to sonatype snapshots
* Added sonatype plugin, and incorporated promote phase into release process

### CODE REVIEW

File: framework/project/Build.scala [hunk 1]
```diff
@@ -52,8 +53,10 @@ object BuildSettings {
     licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0.html")),
 
     ivyLoggingLevel := UpdateLogging.DownloadOnly,
-    resolvers ++= ResolverSettings.playResolvers,
-    resolvers += "Scalaz Bintray Repo" at "http://dl.bintray.com/scalaz/releases", // specs2 depends on scalaz-stream
+    resolvers ++= Seq(
+      Resolver.typesafeRepo("releases"),
+      "Scalaz Bintray Repo" at "http://dl.bintray.com/scalaz/releases"
```
Comment (benmccann) [inline framework/project/Build.scala]: I keep trying to get rid of the http and they keep coming back :-)  Can we make this https please?

https://news.ycombinator.com/item?id=8099713

Comment (jroper) [inline framework/project/Build.scala]: Updated.

**Comment by jroper** [comment]
@play-pull-request-validator

**Comment by jroper** [comment]
@play-pull-request-validator

**Pull request merged**